### ISSUE TITLE
Getting this issue.

### ISSUE THREAD

**Issue by sittim** [oblitum/YouCompleteMe#19]
# Issue Prelude

I added the issue here by accident, can you help?:

https://github.com/Valloric/YouCompleteMe/issues/2490

# Issue Prelude
Have been  using YCM for a long time, but in past two months or so have been getting errors below:

![selection_031](https://cloud.githubusercontent.com/assets/8825037/21622629/8441af68-d1cc-11e6-9527-8ee043ab6732.png)

```
/home/sporty/HydroGuardFW/hg_os~> nvim --version
NVIM 0.2.0-dev
Build type: RelWithDebInfo
Compilation: /usr/bin/x86_64-linux-gnu-gcc -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -D_GNU_SOURCE -I/build/neovim-VogNTX/neovim-0.1.5ubuntu1+git201701021857+3007+22~ubuntu16.04.1/build/config -I/build/neovim-VogNTX/neovim-0.1.5ubuntu1+git201701021857+3007+22~ubuntu16.04.1/src -I/build/neovim-VogNTX/neovim-0.1.5ubuntu1+git201701021857+3007+22~ubuntu16.04.1/.deps/usr/include -I/build/neovim-VogNTX/neovim-0.1.5ubuntu1+git201701021857+3007+22~ubuntu16.04.1/.deps/usr/include -I/build/neovim-VogNTX/neovim-0.1.5ubuntu1+git201701021857+3007+22~ubuntu16.04.1/.deps/usr/include -I/build/neovim-VogNTX/neovim-0.1.5ubuntu1+git201701021857+3007+22~ubuntu16.04.1/.deps/usr/include -I/build/neovim-VogNTX/neovim-0.1.5ubuntu1+git201701021857+3007+22~ubuntu16.04.1/.deps/usr/include -I/build/neovim-VogNTX/neovim-0.1.5ubuntu1+git201701021857+3007+22~ubuntu16.04.1/.deps/usr/include -I/usr/include -I/build/neovim-VogNTX/neovim-0.1.5ubuntu1+git201701021857+3007+22~ubuntu16.04.1/build/src/nvim/auto -I/build/neovim-VogNTX/neovim-0.1.5ubuntu1+git201701021857+3007+22~ubuntu16.04.1/build/include
Compiled by root@lcy01-16

Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

/home/sporty/HydroGuardFW/hg_os~>
```

![selection_032](https://cloud.githubusercontent.com/assets/8825037/21622722/f4aa2528-d1cc-11e6-9ca8-11cd7a4acc33.png)

## Logs

```
2017-01-03 13:48:40,922 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 13:48:47,684 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 13:51:39,575 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 15:07:11,453 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 15:12:42,694 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/completion_request.py", line 60, in RawResponse
    raise MakeServerException( e )
ServerError: RuntimeError: No completions found; errors in the file?
2017-01-03 15:17:24,463 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 15:18:37,475 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 15:27:25,180 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 15:31:11,499 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 15:38:23,052 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 15:43:55,858 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 15:45:08,734 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
2017-01-03 15:46:59,829 - ERROR - Error while handling server response
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 218, in HandleServerException
    yield
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/event_notification.py", line 62, in Response
    self._cached_response = JsonFromFuture( self._response_future )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/autoload/../python/ycm/client/base_request.py", line 188, in JsonFromFuture
    raise MakeServerException( response.json() )
ServerError: RuntimeError: File already being parsed.
```

## STD out log

```
serving on http://127.0.0.1:55782
```

## STD Error log

```
2017-01-03 13:34:35,654 - INFO - Received event notification
2017-01-03 13:34:35,655 - ERROR - No semantic completer exists for filetypes: [u'startify']
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 101, in FiletypeCompletionAvailable
    self.GetFiletypeCompleter( filetypes )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 90, in GetFiletypeCompleter
    current_filetypes ) )
ValueError: No semantic completer exists for filetypes: [u'startify']
2017-01-03 13:34:35,664 - INFO - Received event notification
2017-01-03 13:34:35,665 - INFO - Adding buffer identifiers for file: /home/sporty/HydroGuardFW/hw_1_5/Startify
2017-01-03 13:34:35,667 - ERROR - No semantic completer exists for filetypes: [u'startify']
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 101, in FiletypeCompletionAvailable
    self.GetFiletypeCompleter( filetypes )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 90, in GetFiletypeCompleter
    current_filetypes ) )
ValueError: No semantic completer exists for filetypes: [u'startify']
2017-01-03 13:34:40,680 - INFO - Received health request
2017-01-03 13:34:40,684 - INFO - Received event notification
2017-01-03 13:34:40,684 - ERROR - No semantic completer exists for filetypes: [u'ctrlsf']
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 101, in FiletypeCompletionAvailable
    self.GetFiletypeCompleter( filetypes )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 90, in GetFiletypeCompleter
    current_filetypes ) )
ValueError: No semantic completer exists for filetypes: [u'ctrlsf']
2017-01-03 13:34:40,688 - INFO - Received filetype completion available request
2017-01-03 13:34:40,689 - ERROR - No semantic completer exists for filetypes: [u'ctrlsf']
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 101, in FiletypeCompletionAvailable
    self.GetFiletypeCompleter( filetypes )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 90, in GetFiletypeCompleter
    current_filetypes ) )
ValueError: No semantic completer exists for filetypes: [u'ctrlsf']
2017-01-03 13:34:40,702 - INFO - Received event notification
2017-01-03 13:34:40,702 - INFO - Adding buffer identifiers for file: /home/sporty/HydroGuardFW/hw_1_5/__CtrlSF__
2017-01-03 13:34:40,702 - ERROR - No semantic completer exists for filetypes: [u'ctrlsf']
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 101, in FiletypeCompletionAvailable
    self.GetFiletypeCompleter( filetypes )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 90, in GetFiletypeCompleter
    current_filetypes ) )
ValueError: No semantic completer exists for filetypes: [u'ctrlsf']
2017-01-03 13:34:40,712 - INFO - Received event notification
2017-01-03 13:34:40,713 - INFO - Adding buffer identifiers for file: /home/sporty/HydroGuardFW/hw_1_5/__CtrlSF__
2017-01-03 13:34:40,713 - ERROR - No semantic completer exists for filetypes: [u'ctrlsf']
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 101, in FiletypeCompletionAvailable
    self.GetFiletypeCompleter( filetypes )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 90, in GetFiletypeCompleter
    current_filetypes ) )
ValueError: No semantic completer exists for filetypes: [u'ctrlsf']
2017-01-03 13:34:40,721 - INFO - Received event notification
2017-01-03 13:34:40,721 - ERROR - No semantic completer exists for filetypes: [u'ctrlsf']
Traceback (most recent call last):
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 101, in FiletypeCompletionAvailable
    self.GetFiletypeCompleter( filetypes )
  File "/home/sporty/.config/nvim/plugged/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 90, in GetFiletypeCompleter
    current_filetypes ) )
ValueError: No semantic completer exists for filetypes: [u'ctrlsf']
2017-01-03 13:44:38,130 - INFO - Received health request
2017-01-03 13:54:38,210 - INFO - Received health request
2017-01-03 14:04:38,312 - INFO - Received health request
2017-01-03 14:14:38,420 - INFO - Received health request
2017-01-03 14:24:38,518 - INFO - Received health request
2017-01-03 14:34:38,588 - INFO - Received health request
2017-01-03 14:44:38,651 - INFO - Received health request
2017-01-03 14:54:38,735 - INFO - Received health request
2017-01-03 15:04:38,840 - INFO - Received health request
2017-01-03 15:14:38,884 - INFO - Received health request
2017-01-03 15:24:38,941 - INFO - Received health request
2017-01-03 15:34:39,048 - INFO - Received health request
2017-01-03 15:44:39,120 - INFO - Received health request
2017-01-03 15:54:39,226 - INFO - Received health request
```

**Please complete these steps and check these boxes (by putting an `x` inside
the brackets) _before_ filing your issue:**

- [ ] I have read and understood YCM's [CONTRIBUTING][cont] document.
- [ ] I have read and understood YCM's [CODE_OF_CONDUCT][code] document.
- [ ] I have read and understood YCM's [README][readme], **especially the
  [Frequently Asked Questions][faq] section.**
- [ ] I have searched YCM's issue tracker to find issues similar to the one I'm
  about to report and couldn't find an answer to my problem. ([Example Google
  search.][search])
- [ ] If filing a bug report, I have included the output of `vim --version`.
- [ ] If filing a bug report, I have included the output of `:YcmDebugInfo`.
- [ ] If filing a bug report, I have attached the contents of the logfiles using
  the `:YcmToggleLogs` command.
- [ ] If filing a bug report, I have included which OS (including specific OS
  version) I am using.
- [ ] If filing a bug report, I have included a minimal test case that reproduces
  my issue.
- [ ] **I understand this is an open-source project staffed by volunteers and
  that any help I receive is a selfless, heartfelt _gift_ of their free time. I
  know I am not entitled to anything and will be polite and courteous.**
- [ ] **I understand my issue may be closed if it becomes obvious I didn't
  actually perform all of these steps.**

Thank you for adhering to this process! It ensures your issue is resolved
quickly and that neither your nor our time is needlessly wasted.

# Issue Details

[If filing a bug report, please include **a list of steps** that describe how to
reproduce the bug you are experiencing. Also include test code if relevant.]

[cont]: https://github.com/Valloric/YouCompleteMe/blob/master/CONTRIBUTING.md
[code]: https://github.com/Valloric/YouCompleteMe/blob/master/CODE_OF_CONDUCT.md
[readme]: https://github.com/Valloric/YouCompleteMe/blob/master/README.md
[faq]: https://github.com/Valloric/YouCompleteMe/blob/master/README.md#faq
[search]: https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fgithub.com%2FValloric%2FYouCompleteMe%2Fissues%20python%20mac

**Comment by oblitum** [comment]
I think I just fixed this in the last commit last night. It's strange though that it was not showing up for me, than out of nothing it started to happen. I tried to capture all info I could and put it in the commit description.

Please update and reinstall to check whether it's already solved for you too.

**Comment by oblitum** [comment]
I'm closing as there's no further comments after fix. Feel free to comment.

**Comment by oblitum** [comment]
It was fixed at d7bb3fbc7051f58ad369656996937d5a96895374.
